File: java/src/main/java/com/kittycat/Fibonacci.java
package com.kittycat;

public final class Fibonacci {
    private Fibonacci() {}

    public static long fibonacci(int n) {
        if (n < 0) throw new IllegalArgumentException("n must be non-negative");
        if (n < 2) return n;
        long a = 0, b = 1;
        for (int i = 2; i <= n; i++) {
            long next = a + b;
            a = b;
            b = next;
        }
        return b;
    }

    public static void main(String[] args) {
        int count = args.length > 0 ? Integer.parseInt(args[0]) : 10;
        for (int i = 0; i <= count; i++) {
            System.out.println("fib(" + i + ") = " + fibonacci(i));
        }
    }
}

File: python/fibonacci.py
"""Iterative Fibonacci."""
from __future__ import annotations
import sys


def fibonacci(n: int) -> int:
    if n < 0:
        raise ValueError("n must be non-negative")
    a, b = 0, 1
    for _ in range(n):
        a, b = b, a + b
    return a


def main(argv: list[str]) -> None:
    count = int(argv[1]) if len(argv) > 1 else 10
    for i in range(count + 1):
        print(f"fib({i}) = {fibonacci(i)}")


if __name__ == "__main__":
    main(sys.argv)
